apps/backoffice-v2/src/lib/blocks/hooks/useDocumentBlocks/useDocumentBlocks.tsx

```diff
@@ -264,16 +270,28 @@ export const useDocumentBlocks = ({
                 ),
                 title: `Approval confirmation`,
                 description: <p className={`text-sm`}>Are you sure you want to approve?</p>,
+                content: (
+                  <TextArea
+                    placeholder={'Add a comment'}
+                    value={commentValue}
+                    onChange={event => setCommentValue(event.target.value)}
+                  />
+                ),
                 close: (
                   <div className={`space-x-2`}>
-                    <Button type={'button'} variant={`secondary`}>
+                    <Button
+                      type={'button'}
+                      variant={`secondary`}
+                      onClick={() => setCommentValue('')}
+                    >
                       Cancel
                     </Button>
                     <Button
                       disabled={!canApprove}
                       onClick={onMutateApproveTaskById({
                         taskId: id,
                         contextUpdateMethod: 'base',
+                        comment: commentValue || undefined,
```

Comment:
If we need `|| undefined` isn't it better for the initial value to be empty and for clear to set it back to empty?